redis/src/push_manager.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{PushKind, RedisResult, Value};
+use arc_swap::ArcSwap;
+use std::sync::Arc;
+
+/// Holds information about received Push data
+#[derive(Debug, Clone)]
+pub struct PushInfo {
+    /// Push Kind
+    pub kind: PushKind,
+    /// Data from push message
+    pub data: Vec<Value>,
+
+    /// Connection address to distinguish connections
+    pub con_addr: Arc<String>,
+}
+
+/// Manages Push messages for both tokio and std channels
+#[derive(Clone, Default)]
+pub struct PushManager {
+    sender: Arc<ArcSwap<Option<tokio::sync::mpsc::UnboundedSender<PushInfo>>>>,
+}
+impl PushManager {
+    /// It checks if value's type is Push
+    /// then invokes `try_send_raw` method
+    pub(crate) fn try_send(&self, value: &RedisResult<Value>, con_addr: &Arc<String>) {
```

Comment:
result has to given back to the caller.
```rust
let result = crate::parser::parse_redis_value_async(&mut self.decoder, &mut self.con).await;
self.push_manager.try_send(&result, &self.con_addr);
```

I think there is a miss understanding here

Reply:
fair enough, on re-reading the code and [your comment](https://github.com/redis-rs/redis-rs/pull/898#discussion_r1349151729) I think I understand . This does mean that large values are cloned, even if they'll be dropped later because they're only PUSH messages and not returned as replies to requests. Seems to me this cost can be avoided if we decide to push values instead of references here, instead of keeping the behavior of `read_response`.

If you find this to be complex, this doesn't have to happen on this PR - AFAIS this is completely internal behavior, and we could modify it later without a change in external behavior.

Reply:
Hi, actually the `result`(RedisResult<Value>) is not cloned until it's ready to be sent to a channel. Also read_response should return `clear_active_subscriptions` if we can avoid clear_active_subscriptions with RESP3 then it's possible to make it without any clone.

Reply:
I see that `clear_active_subscriptions` isn't defined for multiplexed connections, or for cluster connections. As mentioned below, I believe that we can drop the `aio::Connection` implementation, since it's likely to be deprecated soon. This means that only the sync `Connection` will have `clear_active_subscriptions` - would that enable us to drop the extra clone for other connections?

Reply:
Hi! Without sync and aio::Connection only thing tries to send something to PushManager is multiplexed_connecton's `send_result` function. In subscribe/unsubscribe the result sent to both push manager and the issuer of the command. It's possible to clone only subscribe/unsubscribe commands then send all values directly to pushmanager.